sdk/lib/atl/atlwin.h

```diff
@@ -320,8 +320,20 @@ class CWindow
         int wndCenterHeight = wndCenterRect.bottom - wndCenterRect.top;
         int wndWidth = wndRect.right - wndRect.left;
         int wndHeight = wndRect.bottom - wndRect.top;
+        int xPos = wndCenterRect.left + ((wndCenterWidth - wndWidth + 1) >> 1);
+        int cxScreen = GetSystemMetrics(SM_CXSCREEN);
+
+        if (xPos + wndWidth > cxScreen)
+        {
+            xPos = cxScreen - wndWidth;
+        }
+        else if (xPos < 0)
+        {
+            xPos = 0;
+        }
+
         return ::MoveWindow(m_hWnd,
-                            wndCenterRect.left + ((wndCenterWidth - wndWidth + 1) >> 1),
+                            xPos,
                             wndCenterRect.top + ((wndCenterHeight - wndHeight + 1) >> 1),
```

Comment:
1: Are you sure that this is the correct place to fix this?
2: If this is the correct place, only the x needs fixing up, not the y?

Reply:
Yes this is the correct place.
Ok your comment makes sense about yPos. Thanks.